__tests__/integration/integration.test.ts

```diff
@@ -0,0 +1,232 @@
+import {makeInputParameters} from '../../src/input-parameters'
+import {
+  Client,
+  ClientConfiguration,
+  Repository
+} from '@octopusdeploy/api-client'
+import {randomBytes} from 'crypto'
+import {CleanupHelper} from './cleanup-helper'
+import {
+  GuidedFailureMode,
+  PackageRequirement,
+  RunbookSnapshotResource,
+  RunCondition,
+  StartTrigger,
+  TenantedDeploymentMode
+} from '@octopusdeploy/message-contracts'
+import {RunConditionForAction} from '@octopusdeploy/message-contracts/dist/runConditionForAction'
+import {RunbookEnvironmentScope} from '@octopusdeploy/message-contracts/dist/runbookEnvironmentScope'
+import {setOutput} from '@actions/core'
+import {OctopusCliWrapper} from '../../src/octopus-cli-wrapper'
+
+const octoExecutable = process.env.OCTOPUS_TEST_CLI_PATH || 'octo' // if 'octo' isn't in your system path, you can override it for tests here
+
+const apiClientConfig: ClientConfiguration = {
+  apiKey:
+    process.env.OCTOPUS_TEST_APIKEY || 'API-XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX',
+  apiUri: process.env.OCTOPUS_TEST_URL || 'http://localhost:8050'
+}
+
+// experimental. Should probably be a custom jest matcher
+// function expectMatchAll(actual: string[], expected: (string | RegExp)[]): void {
+//   expect(actual.length).toEqual(expected.length)
+//   for (let i = 0; i < actual.length; i++) {
+//     const a = actual[i]
+//     const e = expected[i]
+//     if (e instanceof RegExp) {
+//       expect(a).toMatch(e)
+//     } else {
+//       expect(a).toEqual(e)
+//     }
+//   }
+// }
+
+describe('integration tests', () => {
+  const runId = randomBytes(16).toString('hex')
+  const globalCleanup = new CleanupHelper()
+  const localProjectName = `project${runId}`
+  const standardInput = makeInputParameters({
+    project: localProjectName,
+    apiKey: apiClientConfig.apiKey,
+    server: apiClientConfig.apiUri
+  })
+
+  let apiClient: Client
+  beforeAll(async () => {
+    apiClient = await Client.create(apiClientConfig)
+
+    const repository = new Repository(apiClient)
+
+    const lifeCycle = (await repository.lifecycles.all())[0]
+    if (!lifeCycle) throw new Error("Can't find first lifecycle")
+
+    const projectGroup = (await repository.projectGroups.all())[0]
+    if (!projectGroup) throw new Error("can't find first projectGroup")
+
+    const project = await repository.projects.create({
+      Name: localProjectName,
+      LifecycleId: lifeCycle.Id,
+      ProjectGroupId: projectGroup.Id
+    })
+    standardInput.project = project.Id
+    globalCleanup.add(async () => repository.projects.del(project))
```

Comment:
the `del` method returns a Promise already, so should be able to remove the async

`globalCleanup.add(() => repository.projects.del(project))`